### PR TITLE
Refactor : sync 4238 remove dynamo db and dual from tests

### PR DESCRIPTION
**Pull request by taddes**
Push runs integration tests in both scoped Bigtable, dynamo, and dual mode. Given the migration to Bigtable, it is necessary to remove the setup, configuration and test runs against DynamoDB in the integration tests.

 Note:
For now, I have left the many legacy annotated DynamoDB docs. Several were annotated with legacy already. I removed what was related to setup to avoid any confusion, but architectural or systems-related documentation was left in case it has some utility. We can easily drop another PR in the future to remove that, or if someone prefers, I can remove it here.

Acceptance Criteria:

- Remove dependencies related to DynamoDB (boto3, botocore)
- Remove DynamoDB setup at test invocation
- Remove any hidden files or directories (or .gitignore references for build artifacts)
- Remove classes and modules associated with Dynamo setup
- Remove Dynamo as a distinct test mode
- Update docs to remove DynamoDB references and any setup steps

Closes [SYNC-4238](https://mozilla-hub.atlassian.net/browse/SYNC-4238)

[SYNC-4238]: https://mozilla-hub.atlassian.net/browse/SYNC-4238?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -157,22 +151,10 @@ jobs:
...
-            TEST_RESULTS_DIR: workspace/test-results
-      - run:
-          name: Integration tests (Dual Bigtable/DynamoDB)
-          command: make integration-test
           environment:
             RUST_LOG: autopush=debug,autopush_common=debug,autoendpoint=debug,autoconnect=debug,slog_mozlog_json=info,warn
             # PYTEST_ARGS: -sv
-            DB_DSN: dual
-            DB_SETTINGS: tests/integration/dual_test.json
+            DB_DSN: grpc://localhost:8086
```
Comment (jrconlin) [inline .circleci/config.yml:157]: Note for me: 
the `test_integration_all_rust.py` already sets the table names via an env var.

File: docs/src/bigtable-emulation.md [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+# Bigtable Emulation
 
-The bigtable emulator is memory only. No structure persists after
+## Using Bigtable Emulation for Testing
+
+The Bigtable emulator is memory only. No structure persists after
 restart.
 
-## Getting the SDK
+You can also read the testing documentation for these steps
```
Comment (jrconlin) [inline docs/src/bigtable-emulation.md:8]:
```suggestion
You can also read the testing documentation for these steps.
```

**Review by jrconlin — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (taddes) [reply .circleci/config.yml:157]: Yeah good point. Perhaps considering the removal of that in the tests so we're explicit about config might be wise, however I think whatever we pass in via an env var here overrides whatever may be set in the test? Or not?

**Comment by taddes** [comment]
Sorry for need to re-review @jrconlin , had to add a change to `.gitignore` and blew away approval. Sigh Github

**Review by jrconlin — APPROVED** [review]

**Pull request merged**